### PR TITLE
Add TypeDef support, minor tweaks to XML output (to match XSD)

### PR DESCRIPTION
**Pull request by gmarcosb**
Primary change is adding of TypeDef support

See for example `VideoStreamId` @ https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/a3ef8252366959537d75d4e32060d89ca18756f1/src/app_clusters/WebRTC_Provider.adoc

And resulting output with the updates here: https://github.com/project-chip/connectedhomeip/blob/45216a86c1178f94a29688d114263e87321e8352/src/app/zap-templates/zcl/data-model/chip/web-rtc-provider-cluster.xml

Minor tweaks:
1) added the zcl.xsd schema to the output files so that XML editors (e.g. IDEs) are able to check correctness of XML files
2) made some minor fixes to output XML (use name attribute in domain as it's specified in XSD)

### CODE REVIEW

**Comment by gmarcosb** [comment]
@hasty to review, @bzbarsky-apple asked me to reach out regarding adding this support @ https://github.com/project-chip/connectedhomeip/pull/35773

File: zap/generate/configurator.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ func (tg *TemplateGenerator) renderZapTemplate(configurator *zap.Configurator, x
 	ce := x.SelectElement("configurator")
 	if ce == nil {
 		ce = x.CreateElement("configurator")
+		ce.CreateAttr("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance")
+		ce.CreateAttr("xsi:noNamespaceSchemaLocation", "../../zcl.xsd")
 	}
```
Comment (hasty) [inline zap/generate/configurator.go:52]:
```suggestion
		ce = x.CreateElement("configurator")
	}
	ce.CreateAttr("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance")
	ce.CreateAttr("xsi:noNamespaceSchemaLocation", "../../zcl.xsd")
```
I'm guessing you want this added to existing templates as well, but if it's only meant for new templates, ignore this suggestion.

File: matter/spec/cluster.go [hunk 2]
```diff
@@ -90,6 +93,7 @@ func (s *Section) toClusters(d *Doc, entityMap map[asciidoc.Attributable][]types
 		c.AddBitmaps(bitmaps...)
 		c.AddEnums(enums...)
 		c.AddStructs(structs...)
+		c.TypeDefs = append(c.TypeDefs, typedefs...)
```
Comment (hasty) [inline matter/spec/cluster.go:96]: I made changes to the hierarchy resolution code to better disambiguate types with the same name in different parts of the spec. I can update this after merge to do the same for typedefs.

Comment (gmarcosb) [reply matter/spec/cluster.go:96]: Sounds good, thank you!

File: zap/generate/configurator.go (hunk 1 above)
Comment (gmarcosb) [reply zap/generate/configurator.go]: Thank you! I wasn't sure about mass-updating existing ones, even though it is ideal (so all of them properly ref schema), but realized later that re-formatting of generated file seems to be quite common in the matter repo 😅🤣

**Pull request merged**